Translate createElement page (#577)
reactjs/es.reactjs.org: beta/src/content/apis/react/createElement.md

<Intro>

> `createElement` te permite crear un elemento React. Sirve como alternativa a la escritura de [JSX.](/learn/writing-markup-with-jsx)

Reviewer: ```suggestion
`createElement` te permite crear un elemento React. Sirve como alternativa a escribir [JSX.](/learn/writing-markup-with-jsx)
```

---

## Uso {/*usage*/}

> ### Creando un elemento sin JSX {/*creating-an-element-without-jsx*/}

Reviewer: ```suggestion
### Crear un elemento sin JSX {/*creating-an-element-without-jsx*/}
```

}
```

> Los <CodeStep step={3}>children</CodeStep> son opcionales, y puedes pasar tantos como tu necesites (el ejemplo anterior tiene tres hijos). Este código mostrará un encabezado `<h1>` con un saludo. A modo de comparación, aquí está el mismo ejemplo reescrito con JSX:

Reviewer: ```suggestion
Los <CodeStep step={3}>children</CodeStep> son opcionales, y puedes pasar tantos como necesites (el ejemplo anterior tiene tres hijos). Este código mostrará un encabezado `<h1>` con un saludo. A modo de comparación, aquí está el mismo ejemplo reescrito con JSX:
```

}
```

> Para renderizar tu propio componente de React, pasas una función como `Greeting` como el <CodeStep step={1}>type</CodeStep> en lugar de un `'h1'`:

Reviewer: ```suggestion
Para renderizar tu propio componente de React, pasas una función como `Greeting` como el <CodeStep step={1}>type</CodeStep> en lugar de un string como `'h1'`:
```

}
```

[Ver más ejemplos arriba.](#usage)

#### Parámetros {/*parameters*/}

* `type`: El argumento `type` debe ser un tipo de componente de React válido. Por ejemplo, podría ser un string con el nombre de una etiqueta (como `'div'` o `'span'`), o un componente de React (una función, una clase, o un componente especial como [`Fragment`](/apis/react/Fragment)).

> * `props`: El argumento `props` debe ser un objeto o `null`. Si tu le pasas `null`, será tratado igual que un objecto vacio. React creará un elemento con props que coincidan con el `props` que tu has pasado. Ten en cuenta que `ref` y `key` de tu objecto `props` son especiales y lo harán *no* estar disponible como `element.props.ref` y `element.props.key` en el `element` devuelto. Estarán disponibles como `element.ref` y `element.key`.

Reviewer: ```suggestion
* `props`: El argumento `props` debe ser un objeto o `null`. Si tu le pasas `null`, será tratado igual que un objecto vacío. React creará un elemento con props que coincidan con las `props` que has pasado. Ten en cuenta que `ref` y `key` de tu objecto `props` son especiales y lo harán *no* estar disponible como `element.props.ref` y `element.props.key` en el `element` devuelto. Estarán disponibles como `element.ref` y `element.key`.
```

> * **opcional** `...children`: Cero o más nodos. Pueden ser cualquier nodo de React, incluidos Elementos de React, strings, números, [portales](/apis/react-dom/createPortal), nodos vacíos (`null`, `undefined`, `true`, y `false`), y arreglos con nodos de React.

Reviewer: ```suggestion
* **opcional** `...children`: Cero o más nodos. Pueden ser cualquier nodo de React, incluidos Elementos de React, strings, números, [portales](/apis/react-dom/createPortal), nodos vacíos (`null`, `undefined`, `true`, y `false`), y arrays con nodos de React.
```

> #### Returns {/*returns*/}

Reviewer: ```suggestion
#### Devuelve {/*returns*/}
```

`createElement` devuelve un objecto React element con algunas propiedades:

* `type`: El `type` que pasaste.
* `props`: Los `props` que pasaste excepto `ref` y `key`. Si el `type` es un componente con `type.defaultProps` heredado, entonces cualquier `props` que falte o `props` indefinidas obtendrá los valores de `type.defaultProps`.

> * `ref`: El `ref` que pasaste. Si no lo pasaste es, `null`.

Reviewer: ```suggestion
* `ref`: La `ref` que pasaste. Si no la pasaste es, `null`.
```

> * `key`: El `key` que pasaste, forzado a ser string. Si no lo pasaste es, `null`.

Reviewer: ```suggestion
* `key`: La `key` que pasaste, forzada a ser string. Si no la pasaste es, `null`.
```

Por lo general, devolverá el elemento de tu componente o lo convertirá en hijo de otro elemento. Aunque puedes leer las propiedades del elemento, es mejor tratar cada elemento como opaco después de su creación, y solo renderizarlo.

#### Advertencias {/*caveats*/}

> * Debes **tratar elementos React y sus props como [immutables](https://es.wikipedia.org/wiki/Objeto_inmutable)** y nunca cambiar sus contenidos después de la creación. En desarrollo, React [congelaría](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) el elemento devuelto y su propiedad `props` para hacer cumplir esto.

Reviewer: ```suggestion
* Debes **tratar los elementos React y sus props como [inmutables](https://es.wikipedia.org/wiki/Objeto_inmutable)** y nunca cambiar sus contenidos después de creados. En desarrollo, React [congelaría](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) superficialmente el elemento devuelto y sus `props` para hacer cumplir esto.
```

> * Cuando usas JSX, **debe comenzar una etiqueta con una letra mayúscula para representar tu propio componente personalizado.** En otras palabras, `<Something />` es equivalente a `createElement(Something)`, pero `<something />` (minúscula) es equivalente a `createElement('something')` (ten encuenta que es un string, por lo que se tratará como una etiqueta de HTML normal).

Reviewer: ```suggestion
* Cuando usas JSX, **debes comenzar una etiqueta con una letra mayúscula para renderizar tu propio componente personalizado.** En otras palabras, `<Something />` es equivalente a `createElement(Something)`, pero `<something />` (minúscula) es equivalente a `createElement('something')` (ten encuenta que es un string, por lo que se tratará como una etiqueta de HTML normal).
```
